lib/resources/sessions.js

```diff
@@ -38,7 +52,7 @@ module.exports = (service, { User, Session }) => {
       .then(getOrReject(Problem.user.insufficientRights()))
       .then((token) => auth.canOrReject('endSession', token.actor)
         .then(() => token.delete())
-        .then(success))));
+        .then(() => withSessionCookie('null; Expires=Thu, 01 Jan 1970 00:00:00 GMT')(success)))));
```

Comment:
it's still there!

Reply:
Sorry, that wasn't super clear... I'm noticing that before, `success` was immediately invoked, and the promise returned the result of `success`:

```js
.then(success)
```

The new line doesn't seem to immediately invoke `success`. Instead, the promise returns a function that returns the `success` function itself rather than the result of the `success` function.

```js
.then(() => withSessionCookie('null; Expires=Thu, 01 Jan 1970 00:00:00 GMT')(success))
```

Tests are passing, so I assume that `success` is still invoked down the road, but I was just surprised that the `success` function itself was returned rather than its result.

Reply:
oh—yeah, returned functions get called.